alacritty/src/renderer/rects.rs

```diff
@@ -174,11 +174,15 @@ impl RenderLines {
             end.col += 1;
         }
 
+        let color = if flag.contains(Flags::UNDERLINE) || flag.contains(Flags::DOUBLE_UNDERLINE) {
+            cell.underline_color
+        } else {
+            cell.fg
+        };
```

Comment:
Is this done to prevent strikethrough from using this color? That seems quite unfortunate to me, are you aware of why this escape sequence doesn't apply to all lines?

I suppose it's because both strikethrough and underline might be active at the same time?

Generally this just seems concerning because it might cause further escape sequence to complicate things even further, adding yet another color for strikethrough, but I suppose while a red underline for typos is quite common, the strikethrough might not ever have a reason to be a different color than the text itself.

Generally it might be a good idea to add a comment to this stating that strikethrough always uses the cell's color.

Reply:
Yes, the underline color escape sequence should only affect (special) underline color (set with the `<ESC>[4..m` escape sequences), as in KiTTY, iTerm2, VTE.